tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,419 @@
+from ceph.rados.monitor_workflows import MonitorWorkflows
+from tests.rbd.rbd_peristent_writeback_cache import (
+    PersistentWriteAheadLog,
+    fio_ready,
+    get_entity_level,
+)
+from tests.rbd.rbd_utils import initial_rbd_config
+from tests.rbd.test_rbd_persistent_writeback_cache_flush import kill_fio
+from utility.log import Log
+from utility.utils import run_fio
+
+log = Log(__name__)
+
+
+def verify_cache(pwl, config, cache_client, image_spec):
+    """Method to verify rbd persistant cache test.
+
+    Args:
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        cache_client (Node): The client node for cache testing.
+        image_spec (str): The image specification for testing.
+    """
+    log.info("Writing data to the image using FIO jobs and checking cache status")
+    with parallel() as p:
+        p.spawn(run_fio, **fio_ready(config, cache_client))
+        p.spawn(
+            pwl.check_cache_file_exists,
+            image_spec,
+            config["fio"].get("runtime", 120),
+            **config,
+        )
+
+
+def restart_service_with_cache_test(
+    service_name, rados_obj, pwl, config, cache_client, image_spec
+):
+    """
+    Restart a Ceph service and verify the persistent write cache.
+
+    Args:
+        service_name (str): The name of the Ceph service to restart.
+        rados_obj (RadosOrchestrator): The RadosOrchestrator instance.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        cache_client (Node): The client node for cache testing.
+        image_spec (str): The image specification for testing.
+    """
+    log.info(f"Test to restarting {service_name} service along with pwl cache test")
+    with parallel() as p:
+        p.spawn(rados_obj.restart_daemon_services, service_name)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def restart_osd_service_with_cache_test(
+    cache_client, rados_obj, pwl, config, image_spec
+):
+    """
+    Restart OSD service and verify the persistent write cache.
+
+    Args:
+        cache_client (Node): The client node for cache testing.
+        rados_obj (RadosOrchestrator): The RadosOrchestrator instance.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        image_spec (str): The image specification for testing.
+    """
+    log.info("Test to restart OSD service along with pwl cache test")
+
+    def restart_osd():
+        cmd = "ceph orch ls osd"
+        out = cache_client.exec_command(cmd=cmd, output=True)
+
+        # Regular expression to extract the first OSD name
+        osd_daemon = re.search(r"osd\.(\S+)", str(out))
+        cmd = f"ceph orch restart {osd_daemon[0]}"
+        cache_client.exec_command(cmd=cmd, output=True)
+
+    with parallel() as p:
+        p.spawn(restart_osd)
```

Comment:
This implementation still doesn't catch any return value and throw exception right? As we discussed.